shogun-boot/src/main/resources/templates/index.html

```diff
@@ -42,106 +46,184 @@
+          background-color: rgba(233, 236, 239, 1);
+        }
+
+        .app-buttons {
+          display: flex;
+          justify-content: end;
+          padding: 5px;
+        }
+
+        .app-info {
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+          justify-content: center;
+        }
+      </style>
+    </template>
+    <!-- web component template ends -->
+
+    <div class="container px-4 py-5">
+      <h2 class="pb-2">Welcome to SHOGun</h2>
+      <div style="display: none" id="keycloak-host" th:text="${@environment.getProperty('KEYCLOAK_HOST')}"></div>
+      <a href="/admin">
+        <button id="admin-panel-btn" class="btn btn-primary">
+          Open Admin Panel <i class="fas fa-lock"></i>
+        </button>
+      </a>
+      <h3 class="pb-2 pt-4">Applications</h3>
+      <div class="row row-cols-1 row-cols-md-3 g-4 apps" />
     </div>
 
   </main>
 
-  <footer class="container d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
+  <footer class="container d-flex justify-content-between align-items-center py-3 my-4 border-top">
     <p class="col-md-4 mb-0 text-muted">
       &copy; 2021 - present <a href="https://www.terrestris.de">terrestris GmbH & Co. KG</a>
     </p>
-
-    <ul class="nav col-md-4 justify-content-end">
-      <li class="nav-item">
-        <a href="#" class="nav-link px-2 text-muted" th:text="${'Version: ' + version}" />
-      </li>
-    </ul>
+    <p class="col-md-3 mb-0 text-muted">
+      <i class="fas fa-project-diagram"></i>
+      <a href="./swagger-ui/index.html">Swagger / OpenAPI</a>
+    </p>
+    <p class="col-md-3 mb-0 text-muted">
+      <i class="fas fa-lock"></i>
+      <a href="./graphiql">GraphiQL</a>
+    </p>
+    <p class="col-md-3 mb-0 text-muted">
+      <i class="fas fa-external-link-alt"></i>
+      <a href="https://github.com/terrestris/shogun">GitHub</a>
+    </p>
   </footer>
 
-  <script src="./assets/lib/bootstrap/js/bootstrap.bundle.min.js"></script>
+  <script th:inline="javascript">
+    window.onload = async () => {
+      const keycloakHost = [[${@environment.getProperty('KEYCLOAK_HOST')}]];
+      const keycloak = new Keycloak({
+        url: `https://${keycloakHost}/auth`,
+        realm: 'SHOGun',
+        clientId: 'shogun-client'
+      });
+      try {
+        const authenticated = await keycloak.init({
+          onLoad: 'check-sso'
+        });
+      } catch (error) {
+        console.error(error);
+        return;
+      }
+
+      // Register login/logout actions
+      document.querySelector('#login-btn').addEventListener('click', () => {
+        keycloak.login();
+      });
+      if (keycloak.authenticated) {
+        document.querySelector('#logout-btn').hidden = false;
+        document.querySelector('#login-btn').hidden = true;
+        document.querySelector('#logout-btn').addEventListener('click', () => {
+          keycloak.logout();
+        });
```

Comment:
The event registration might be moved outside the condition (as with the login button).